http-server/src/main/java/org/triplea/modules/chat/event/processing/PlayerIsMutedMessage.java

```diff
@@ -4,23 +4,34 @@
 import java.time.Clock;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.function.BiFunction;
+import java.util.function.Function;
+import lombok.Builder;
 import lombok.experimental.UtilityClass;
 
 @UtilityClass
 class PlayerIsMutedMessage {
+  private static Function<Instant, String> muteDurationFormatter =
+      MuteDurationRemainingCalculator.builder().build();
 
+  String build(final Instant muteExpiry) {
+    return "You have been muted, expiring in: " + muteDurationFormatter.apply(muteExpiry);
+  }
+
+  /**
+   * Calculates a string of how many minutes are remaining in a mute until expired, or if less than
+   * a minute then how many seconds are left.
+   */
   @VisibleForTesting
-  static final BiFunction<Clock, Instant, String> muteDurationRemainingToString =
-      (clock, muteExpiry) -> {
-        final long minutes = Duration.between(clock.instant(), muteExpiry).toMinutes();
-        return minutes > 0
-            ? minutes + " minutes"
-            : Duration.between(clock.instant(), muteExpiry).toSeconds() + " seconds";
-      };
+  @Builder
+  static class MuteDurationRemainingCalculator implements Function<Instant, String> {
+    @Builder.Default private Clock clock = Clock.systemUTC();
 
-  String build(final Instant muteExpiry) {
-    return "You have been muted, expiring in: "
-        + muteDurationRemainingToString.apply(Clock.systemUTC(), muteExpiry);
+    @Override
+    public String apply(final Instant muteExpiry) {
+      final long minutes = Duration.between(clock.instant(), muteExpiry).toMinutes();
+      return minutes > 0
+          ? minutes + " minutes"
+          : Duration.between(clock.instant(), muteExpiry).toSeconds() + " seconds";
```

Comment:
side-note: We could store the result of `Duration.between(clock.instant(), muteExpiry)` in a variable instead of potentially calculating it twice, but just a minor detail